kuma/users/tests/test_views.py

```diff
@@ -1132,75 +1131,44 @@ def test_email_addresses(self):
         assert response.status_code == 200
         assert_no_cache_header(response)
-        assert private_email not in response.context
-        email_address = response.context["email_addresses"]
-
-        # first check if the public email address has been found
-        assert public_email in email_address
-        assert email_address[public_email] == {
-            "verified": False,
-            "email": public_email,
-            "primary": False,
-        }
-        # then check if the private and verified-at-GitHub email address
-        # has been found
-        assert private_email in email_address
-        assert email_address[private_email] == {
-            "verified": True,
-            "email": private_email,
-            "primary": True,
-        }
-        # then check that the invalid email is not present
-        assert invalid_email not in email_address
-        # then check if the radio button's default value is the primary email
-        # address
-        assert response.context["form"].initial["email"][1] == private_email
+        doc = pq(response.content)
+
+        # The hidden input should display the primary verified email
+        assert doc.find('input[name="email"]').val() == email_data[1]["email"]
+        # But whatever's in the hidden email input is always displayed to the user
+        # as "plain text". Check that that also is right.
+        assert doc.find("#email-static-container").text() == email_data[1]["email"]
 
         unverified_email = "o.ctocat@gmail.com"
         data = {
             "website": "",
             "username": "octocat",
-            "email": SignupForm.other_email_value,  # = use other_email
-            "other_email": unverified_email,
+            "email": email_data[1]["email"],
             "terms": True,
         }
         assert not EmailAddress.objects.filter(email=unverified_email).exists()
         response = self.client.post(self.signup_url, data=data)
         assert response.status_code == 302
         assert_no_cache_header(response)
-        unverified_email_addresses = EmailAddress.objects.filter(email=unverified_email)
-        assert unverified_email_addresses.exists()
-        assert unverified_email_addresses.count() == 1
-        assert unverified_email_addresses[0].primary
-        assert not unverified_email_addresses[0].verified
 
-    def test_email_addresses_with_no_public(self):
-        profile_data = self.github_profile_data.copy()
-        profile_data["email"] = None
-        email_data = self.github_email_data[:]
-        private_email = "octocat.private@example.com"
-        email_data[0]["email"] = private_email
-        self.github_login(profile_data=profile_data, email_data=email_data)
-        response = self.client.get(self.signup_url)
-        assert response.status_code == 200
-        assert_no_cache_header(response)
-        assert response.context["form"].initial["email"] == private_email
-
```

Comment:
Death to all tests that test implementation details! It doesn't matter HOW the form works. Just that it works and the inputs match the expected outputs.